### PR TITLE
Add Maven archetype to generate applications

### PR DESCRIPTION
**Pull request by oyving**
@aressem - Did I add a new module in the right way?
@tokle - Did I create an archetype that looks sane?

### CODE REVIEW

File: archetypes/application-java/pom.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>com.yahoo.vespa</groupId>
+    <artifactId>parent</artifactId>
+    <version>7-SNAPSHOT</version>
+    <relativePath>../../parent/pom.xml</relativePath>
+  </parent>
+ 
+  <groupId>ai.vespa.cloud</groupId>
```
Comment (tokle) [inline archetypes/application-java/pom.xml]: we can't use this groupid (at least not yet)

**Comment by aressem** [comment]
Looks good if this is only going to be published for open source / Maven Central.

File: archetypes/application-java/pom.xml (hunk 1 above)
Comment (oyving) [reply archetypes/application-java/pom.xml]: Fixed

**Comment by kkraune** [comment]
I don't want to be the party killer but why?

Last time we had archetypes, they silently failed, but no one noticed as they were not in use - only one more thing we had to maintain, more documentation, more tests etc.

So I killed all of them

It is fully possible I am mistaken here and there is a good case for it - but I will only approve it as long as it is fully documented and tested and followed up every day. _And_ there is a good use case for having it (won't users just copy a sample app?)

**Comment by aressem** [comment]
Is this something that should be brought up on this weeks architect meeting ?

**Comment by bratseth** [comment]
Yes

**Pull request closed without merging**